hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -3597,10 +3599,22 @@ File replaceOMDBWithCheckpoint(long lastAppliedIndex, File oldDB,
     String dbBackupName = OzoneConsts.OM_DB_BACKUP_PREFIX +
         lastAppliedIndex + "_" + System.currentTimeMillis();
     File dbDir = oldDB.getParentFile();
-    File dbBackup = new File(dbDir, dbBackupName);
 
+    // Backup the active fs and snapshot dirs
+    File dbBackupDir = new File(dbDir, dbBackupName);
+    if (!dbBackupDir.mkdirs()) {
+      throw new IOException("Failed to make db backup dir: " +
+          dbBackupDir);
+    }
+    File dbBackup = new File(dbBackupDir, oldDB.getName());
+    File dbSnapshotsDir = new File(dbDir, OM_SNAPSHOT_DIR);
+    File dbSnapshotsBackup = new File(dbBackupDir, OM_SNAPSHOT_DIR);
     try {
       Files.move(oldDB.toPath(), dbBackup.toPath());
+      if (dbSnapshotsDir.exists()) {
+        Files.move(dbSnapshotsDir.toPath(),
+            dbSnapshotsBackup.toPath());
+      }
     } catch (IOException e) {
```

Comment:
This is unnecessary catch. Doing nothing other than double logging and re-throwing the same exception.